dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,114 @@
+# The CI Service in Hail Batch
+
+Hail Batch includes a CI service which has three jobs:
+
+- Runs tests against pull requests
+- Merges PRs to the `main` branch
+- Deploys services to the live infrastructure
+
+It does all of these things only if certain conditions are met.
+
+## Deployment
+
+The CI service itself is deployed as a Kubernetes service in the Hail Batch cluster. See 
+[architecture diagram](../Hail%20Batch%20Architectural%20Diagram.png).
+
+As part of its configuration, CI must be configured with an access token allowing it to operate on 
+behalf of a github account called hail-ci-robot.
+
+To make CI work it needs to receive update triggers from the github repository. These are
+configured within the github repository itself.
+
+## Running tests against pull requests
+
+The CI service is constantly monitoring for new PRs in the `hail` repository. 
+
+When a new PR is opened, or an existing PR changes, hooks are triggered which alert the CI service. 
+The CI service will then:
+
+- Check for validity
+  - The author must be a known developer (or else a commit SHA can be individually approved via the CI service UI)
+  - The PR must be against the `main` branch
```

Comment:
This is true by configuration. `HAIL_WATCHED_BRANCHES` variable in `ci/deployment.yaml` controls this. We could add more than just `main` to this watchlist.

I'm agreeing with what you're saying, just perhaps being pedantic which might be useful if anyone wants to see where this is configured.

Reply:
I think this extra context is super helpful. IMO it's worth including both "how we have it set up" and "what can be configured"

Reply:
This also makes it sound like CI is more poll-based than trigger-based? I don't have enough permissions in the repo to check what hooks we have set up so some of this is speculative (I should probably have done / go ahead and do a better job of highlighting guesswork)

Reply:
In fact, the advice from Daniel that "deleting CI's most recent deploy run will make it run again" seems like it _must_ be polling based. I'll look into this today

Reply:
It's a bit of both. Updated the doc with that info